seed/seed20.go

```diff
@@ -82,6 +84,78 @@ type seed20 struct {
 	essentialSnapsNum int
 }
 
+// Copy implement Copier interface.
+func (s *seed20) Copy(seedDir string, label string) (err error) {
+	srcSystemDir, err := filepath.Abs(s.systemDir)
+	if err != nil {
+		return err
+	}
+
+	if label == "" {
+		label = filepath.Base(srcSystemDir)
+	}
+
+	destSeedDir, err := filepath.Abs(seedDir)
+	if err != nil {
+		return err
+	}
+
+	if err := os.Mkdir(filepath.Join(destSeedDir, "systems"), 0755); err != nil && !errors.Is(err, fs.ErrExist) {
+		return err
+	}
+
+	destSystemDir := filepath.Join(destSeedDir, "systems", label)
+	if osutil.FileExists(destSystemDir) {
+		return fmt.Errorf("cannot create system: system %q already exists at %q", label, destSystemDir)
+	}
+
+	// TODO: is this a good idea? we won't be able to clean up asserted snaps
+	// that are copied, but i think that is ok
+	defer func() {
+		if err != nil {
+			os.RemoveAll(destSystemDir)
+		}
+	}()
+
+	// copy all files (including unasserted snaps) from the seed to the
+	// destination
+	err = filepath.Walk(srcSystemDir, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		destPath := filepath.Join(destSeedDir, "systems", label, strings.TrimPrefix(path, srcSystemDir))
+		if info.IsDir() {
+			return os.Mkdir(destPath, info.Mode())
+		}
+
+		return osutil.CopyFile(path, destPath, osutil.CopyFlagDefault)
+	})
+	if err != nil {
+		return err
+	}
+
+	// copy the asserted snaps that the seed needs
+	for _, sn := range s.snaps {
+		// unasserted snaps are already copied above, skip them
+		if sn.ID() == "" {
+			continue
+		}
+
+		destPath := filepath.Join(destSeedDir, "snaps", filepath.Base(sn.Path))
+
+		if err := os.MkdirAll(filepath.Dir(destPath), 0755); err != nil {
```

Comment:
we might want to try to do this only once

Reply:
I'm not sure I follow. `destPath` changes each loop, so we are only doing this once, I think?

Reply:
Oh no, I can't read. I see.